### PR TITLE
Ks 2022 04 comment notification statement serializer

### PR DESCRIPTION
**Pull request by Rineee**
Left the old apis as they are for now as we might still want to use them.

The viewset is available under /api/userdashboard/moderation/$project_pk/comments/ and I added routes to archive/unarchive,
they are /api/userdashboard/moderation/$project_pk/comments/$comment_pk/archive and /unarchive, sending GET to these will automatically archive/unarchive all notifications of the respective comment.

To Do:

- [ ] - integrate moderator statement api (maybe just serialize url and leave api as it is?)
- [x] - filtering, this includes annotations I think
- [x] - sorting, also requires annotations (this is an own story, see https://taiga.liqd.net/project/mkay-kosmo/us/114?no-milestone=1)
- [x] - connect to frontend and remove old stuff
- [ ] - maybe routes to view classifications as we already have the serializers/apis?
- [ ] - permissions

### CODE REVIEW

File: apps/userdashboard/serializers.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    class Meta:
+        model = Comment
+        fields = ['category_counts', 'comment', 'comment_url',
+                  'last_edit', 'pk', 'has_pending_notifications', 'is_blocked',
+                  'is_moderator_marked', 'moderator_statement',
+                  'time_of_last_notification', 'user_image', 'user_name',
+                  'user_profile_url']
+
+    # helper
+    def has_pending_and_archived_notifications(self, comment):
```
Comment (fuzzylogic2000) [inline apps/userdashboard/serializers.py:36]: I think if we only used internally, there is the convention to call it `_has_pending_and_archived_notifications`.

Comment (fuzzylogic2000) [reply apps/userdashboard/serializers.py:36]: Same for the other helpers.

**Review by fuzzylogic2000 — COMMENTED** [review]
Supernice! I couldn't find anything wrong with the API, only a naming convention thingy.

File: apps/userdashboard/serializers.py (hunk 1 above)
Comment (Rineee) [reply apps/userdashboard/serializers.py:36]: Oh yes, good point, will change that!

**Review by phillimorland — APPROVED** [review]
reads very nicely! and it works :)

**Review by fuzzylogic2000 — COMMENTED** [review]
Supercool!

**Pull request merged**